cmd/initialize_with_genesis.cpp

```diff
@@ -0,0 +1,253 @@
+
+static void check_rlp_err(rlp::DecodingResult err) {
+    if (err != rlp::DecodingResult::kOk) {
+        throw err;
+    }
+}
+
+static intx::uint256 convert_string_to_uint256(std::string const& str) {
+    intx::uint256 res{0};
+    for(size_t i = 0; i < str.size(); i++) {
+        auto current_digit{str.at(i) - '0'};
+        res += current_digit;
+        res *= 10;
+    }
+    return res/10;
+}
+
+static bool is_hex(std::string const& s) {
+  return s.compare(0, 2, "0x") == 0
+      && s.size() > 2
+      && s.find_first_not_of("0123456789abcdefABCDEF", 2) == std::string::npos;
+}
+
+static uint64_t get_uint_from_field(std::string const& field) {
+    int form = is_hex(field)? 16 : 10;
+    uint64_t res = strtoull(field.c_str(), nullptr, form);
+    if (errno == EINVAL)
+    {
+        SILKWORM_LOG(LogLevel::Error) << field << " not a valid number." << std::endl;
+        throw;
+    }
+    else if (errno == ERANGE)
+    {
+        SILKWORM_LOG(LogLevel::Error) << field << " must be uint64" << std::endl;
+        throw;
+    }
+    return res;
+}
+
+int main(int argc, char* argv[]) {
+    namespace fs = boost::filesystem;
+    int chain_id{-1};
+    CLI::App app{"Initializes database with genesis json file"};
+
+    std::string out;
+    std::string genesis;
+    app.add_option("--out", out, "Path to the newly created chaindata", true);
+
+    app.add_option("--genesis", genesis, "Path to the genesis json file", true)
+        ->check(CLI::ExistingDirectory);
+
+    app.add_option("--chainid", chain_id, "Specify id of the chain to generate");
+
+    CLI11_PARSE(app, argc, argv);
+
+    // Check flags
+    if (fs::exists(out)) {
+        SILKWORM_LOG(LogLevel::Error) << out << " already exist." << std::endl;
+        return -1;
+    }
+
+    if (!fs::exists(genesis) && chain_id <= 0) {
+        SILKWORM_LOG(LogLevel::Error) << "genesis was not found." << std::endl;
+        return -1;
+    }
+    // We create the chaindata directory
+    fs::create_directories(out);
+
+    // We Initialize the database and open it
+    lmdb::DatabaseConfig db_config{out};
+    db_config.set_readonly(false);
+    std::shared_ptr<lmdb::Environment> env{lmdb::get_env(db_config)};
+    std::unique_ptr<lmdb::Transaction> txn{env->begin_rw_transaction()};
+    // We create all tables
+    db::table::create_all(*txn);
```

Comment:
I'd move the creation of all tables **after** all sanity checks on genesis config are made.